terra-jupyter-dev-base/Dockerfile

```diff
@@ -0,0 +1,163 @@
+ && sudo -i \
+    echo "deb http://security.ubuntu.com/ubuntu/ bionic main" >> /etc/apt/sources.list \
+    sudo apt update libexempi3 \
+ && sudo -i \
+    echo "deb http://us.archive.ubuntu.com/ubuntu/ bionic universe" >> /etc/apt/sources.list \
+    sudo apt update libv8-3.14-dev \
+ && apt-get update && apt-get install -yq --no-install-recommends \
+    # gnupg requirement
+    dirmngr \
+    gnupg \
+    # curl requirement
+    curl \
+    ca-certificates \
+    # useful utilities for debugging within the docker
+    nano \
+    procps \
+    # extras \
+    wget \
+    bzip2 \
+    # install script requirements
+    locales \
+    # for ssh-agent and ssh-add
+    keychain \
+    # git
+    git \
+ # Uncomment en_US.UTF-8 for inclusion in generation
+ && sed -i 's/^# *\(en_US.UTF-8\)/\1/' /etc/locale.gen \
+ # Generate locale
+ && locale-gen \
+ && apt-get clean \
+ && rm -rf /var/lib/apt/lists/* \
+# install miniconda to $CONDA_DIR
+ && curl -so $HOME/miniconda.sh https://repo.anaconda.com/miniconda/Miniconda3-py37_4.12.0-Linux-x86_64.sh \
+ && chmod +x $HOME/miniconda.sh \
+ && $HOME/miniconda.sh -b -p $CONDA_DIR \
+ && rm $HOME/miniconda.sh \
+# slim install of python packages that have a gcc dependency (cleanup included)
+ && apt-get update && apt-get install -y --no-install-recommends \
+    checkinstall \
+    build-essential \
+    zlib1g-dev \
+    # pip requirements
+    libssl-dev \
+    libbz2-dev \
+    libreadline-dev \
+    libsqlite3-dev \
+    llvm \
+    libncurses5-dev \
+    libncursesw5-dev \
+    tk-dev \
+    libffi-dev \
+    liblzma-dev \
+    python-openssl \
+    libexempi3 \
+    libv8-3.14-dev \
+ && rm -rf /var/lib/apt/lists/* \
+ && pip3 install firecloud  \
+ && pip3 install terra-notebook-utils \
+ && apt-get purge -y --auto-remove \
+    checkinstall \
+    build-essential \
+    zlib1g-dev \
+    # pip requirements
+    libssl-dev \
+    libbz2-dev \
+    libreadline-dev \
+    libsqlite3-dev \
+    llvm \
+    libncurses5-dev \
+    libncursesw5-dev \
+    tk-dev \
+    libffi-dev \
+    liblzma-dev \
+    python-openssl \
+    libexempi3 \
+    libv8-3.14-dev \
+# Install jupyter and some necessary python packages
+ && pip3 -V \
+ # For gcloud alpha storage support.
+ && pip3 install google-crc32c --target /usr/lib/google-cloud-sdk/lib/third_party \
+ && git clone --branch cromshell_2.0 --single-branch \
+    https://github.com/broadinstitute/cromshell.git $HOME/cromshell \
+ # tmp hack min-5
+ # I'm not installing jupyterlab and I can't update init-actions.sh to not access it
+ && mkdir -p /usr/local/share/jupyter/lab \
+ # When we upgraded from jupyter 5.7.8 to 6.1.1, we broke terminal button on terra-ui.
+ # Hence, make sure to manually test out "launch terminal" button (the button in the green bar next to start and stop buttons)
+ # to make sure we don't accidentally break it every time we upgrade notebook version until we figure out an automation test for this
+ && pip3 install notebook \
```

Comment:
Do we want to pin down the versions? And/or have a `requirements.txt`?

Reply:
requirements.txt would be a nice addition!

Reply:
Probably makes sense to pin them